### PR TITLE
OpInfo: reshape, reshape_as and minor clean-up

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: #54261

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -5463,33 +5463,32 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
-           dtypesIfCPU=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
-           dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
-           dtypesIfROCM=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           dtypes=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
            supports_out=False,
            skips=(
                # Because view_as does not have a function variant.
                SkipInfo('TestCommon', 'test_variant_consistency_jit'),),
-           sample_inputs_func=sample_inputs_view_as,
+           sample_inputs_func=sample_inputs_view_as_reshape_as,
```
Comment (anjali411) [inline torch/testing/_internal/common_methods_invocations.py:5491]: `view_as` and `reshape_as` are not exactly same, so I think we should add an `OpInfo` for `reshape_as` as well?

**Review by anjali411 — APPROVED** [review]

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py:5491]: Both have different OpInfo entries. However both share the `sample_inputs_func`.

Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py:5491]: Actually you are right. My bad! Forgot to add entry for `reshape_as`. Thanks!

Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py:5500]: @anjali411
Updated!
Thank you very much!

**Comment by kshitij12345** [comment]
@anjali411 this should be good to go now. Thanks!

**Comment by kshitij12345** [comment]
@anjali411 Gentle Ping :)

**Comment by anjali411** [comment]
could you rebase this PR?

**Pull request closed without merging**